### PR TITLE
introduce atomic for the matcher (routing table) pointer

### PR DESCRIPTION
**Pull request by aryszka**

### CODE REVIEW

**Comment by lmineiro** [comment]
lgtm

File: routing/routing.go [hunk 1]
```diff
@@ -130,10 +131,22 @@ func feedMatchers(updateBuffer int, current *matcher) (chan<- *matcher, <-chan *
 // Initializes a new routing instance, and starts listening for route
 // definition updates.
 func New(o Options) *Routing {
+	r := &Routing{}
 	initialMatcher, _ := newMatcher(nil, MatchingOptionsNone)
-	matchersIn, matchersOut := feedMatchers(o.UpdateBuffer, initialMatcher)
```
Comment (mo-gr) [inline routing/routing.go]: `feedMatchers` is now dead code and can be removed

**Comment by mo-gr** [comment]
lgtm

File: routing/routing.go (hunk 1 above)
Comment (aryszka) [inline routing/routing.go]: right, updating in a sec

**Pull request merged**